### PR TITLE
chore(react-examples): replace storybook deprecated api with static stories

### PR DESCRIPTION
**Pull request by dmytrokirpa**
## Previous Behavior

Several deprecated Storybook APIs have been used to dynamically generate and incorporate `react-examples` stories.

## New Behavior

1. Updated deprecated APIs used in `.storybook/preview.js` with their current counterparts:

- `addDecorator(...)` => `export const decorators = [...]`
- `addParameters(...)` => `export const parameters = [...]`
- `configure(module, loadStories)` => `stories: getStories()` in `.storybook/main.js`

2. Added default exports into existing stories for CSF compatibility.
3. Implemented static story files that export components from `./[ComponentName].[Variant].Example`, preserving the existing logic but removing dynamic story generation. For instances where React class components are used, direct re-exporting isn't feasible; hence, they are wrapped and exported as functional components, such as `export const Component = () => <ComponentExample />`.

No changes have been made to the actual `.doc.tsx` or `.Example.tsx` files.

## Related Issue(s)

- Contributes to #30858 
- Blocks #32018

### CODE REVIEW

File: packages/react-examples/.storybook/main.js [hunk 1]
```diff
@@ -48,4 +41,19 @@ const config = {
   },
 };
 
+function getStories() {
+  const packageName = path.basename(__dirname);
+
+  if (!fs.existsSync(path.resolve(__dirname, '../src', packageName))) {
```
Comment (dmytrokirpa) [inline packages/react-examples/.storybook/main.js:47]: This logic is ported from the https://github.com/microsoft/fluentui/pull/32074/files#diff-ab3c9c8aaeafc69909155f590e3326d509375e36844d88e2b29c8e56f78d4a3eL12

File: packages/react-examples/.storybook/main.js [hunk 2]
```diff
@@ -48,4 +41,19 @@ const config = {
...
+    throw new Error(`Package ${packageName} does not have examples!`);
+  }
+
+  // For @fluentui/react's storybook, also include examples from @fluentui/react-focus
+  if (packageName === 'react') {
```
Comment (dmytrokirpa) [inline packages/react-examples/.storybook/main.js:52]: this came from https://github.com/microsoft/fluentui/pull/32074/files#diff-e11bfa97318c042abbf9cd46418de9afda2274a5e8d873e200ee2a5e8bec9832L64

**Comment by fabricteam** [comment]
## 📊 Bundle size report

✅ No changes found
<div id="bundleSizeComment9423"/>

**Review by AtishayMsft — APPROVED** [review]
Approving from charting package side

**Review by spmonahan — APPROVED** [review]

**Pull request merged**